source/postprocess/crystal_preferred_orientation.cc

```diff
@@ -389,11 +389,14 @@ namespace aspect
               std::vector<std::vector<std::array<double,3>>> weighted_euler_angles;
 
               weighted_euler_angles.resize(n_minerals);
+              weighted_rotation_matrices.resize(n_minerals);
               std::vector<std::vector<double>> volume_fractions_grains(n_minerals,std::vector<double>(n_grains,-1.));
               for (unsigned int mineral = 0; mineral < n_minerals; ++mineral)
                 {
                   for (unsigned int i_grain = 0; i_grain < n_grains; ++i_grain)
                     {
+                      weighted_euler_angles[mineral].resize(n_grains);
+                      weighted_rotation_matrices[mineral].resize(n_grains);
```

Comment:
Out of curiosity, why was this necessary? Did you get error messages/segmentation faults without resizing?

Reply:
Yes.. Had a similar issue on the add_drexpp_V3 branch and this helped so just used it here and it works.

Reply:
This points to a bigger problem. If the code previously had a bug whereby you'd get error messages without the `resize()` calls, then there are not enough tests that test this code. Would you be able to write an input file that triggers the error before, and that is fixed with the addition of the code here?